src/Sylius/Component/Core/Factory/ChannelFactory.php

```diff
@@ -0,0 +1,53 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Component\Core\Factory;
+
+use Sylius\Component\Core\Model\ChannelInterface;
+use Sylius\Component\Resource\Factory\FactoryInterface;
+
+class ChannelFactory implements ChannelFactoryInterface
+{
+    /** @var FactoryInterface */
+    private $decoratedFactory;
+
+    public function __construct(FactoryInterface $decoratedFactory)
+    {
+        $this->decoratedFactory = $decoratedFactory;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function createNew(): ChannelInterface
+    {
+        /** @var ChannelInterface $channel */
+        $channel = $this->decoratedFactory->createNew();
+        $channel->setTaxCalculationStrategy('order_items_based');
+
+        return $channel;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function createNamed(string $name): ChannelInterface
+    {
+        /** @var ChannelInterface $channel */
+        $channel = $this->decoratedFactory->createNew();
+        $channel->setTaxCalculationStrategy('order_items_based');
+        $channel->setName($name);
+
+        return $channel;
+    }
```

Comment:
```suggestion
class ChannelFactory implements ChannelFactoryInterface
{
    /** @var FactoryInterface */
    private $decoratedFactory;
    
    /** @var string */
    private $defaultCalculationStrategy;

    public function __construct(FactoryInterface $decoratedFactory, string $defaultCalculationStrategy)
    {
        $this->decoratedFactory = $decoratedFactory;
        $this->defaultCalculationStrategy = $defaultCalculationStrategy;
    }

    /**
     * {@inheritdoc}
     */
    public function createNew(): ChannelInterface
    {
        /** @var ChannelInterface $channel */
        $channel = $this->decoratedFactory->createNew();
        $channel->setTaxCalculationStrategy('order_items_based');

        return $channel;
    }

    /**
     * {@inheritdoc}
     */
    public function createNamed(string $name): ChannelInterface
    {
        $channel = $this->createNew();
        $channel->setName($name);

        return $channel;
    }
```

Reply:
That is probably what @pamil meant. And I agree with that.

Reply:
@mamazu proposal makes sense. However, I need to ask one more thing. Do we need `createNamed` in the first place? It seems to be unused in this scope and therefore we should remove it in the first place IMHO. Other than this, its good to go.

Reply:
Agreed. Because then we also don't need an extra interface for it.

Reply:
Thanks for feedbacks, i'll change it :)

Reply:
Hi,

I use `defaultCalculationStrategy` as parameter of ChannelFactory as proposed by @mamazu but i have keep function `createNamed()` because she is use in

https://github.com/Sylius/Sylius/blob/2875e9e2549eca13d53bc3cd3b6617da7d44ac04/src/Sylius/Component/Core/spec/Test/Services/DefaultChannelFactorySpec.php#L68

What do you think ?

Reply:
Well the `createNamed` doesn't really fit the style of Sylius. But other than that it looks good.